main.tf

```diff
@@ -1,13 +1,192 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled = module.this.enabled
+}
+
+resource "aws_ssm_maintenance_window" "scan_window" {
+  count    = local.enabled ? 1 : 0
+  name     = "module.this.id"
+  schedule = var.scan_maintenance_window_schedule
+  duration = var.maintenance_window_duration
+  cutoff   = var.maintenance_window_cutoff
+}
+
+resource "aws_ssm_maintenance_window_task" "task_scan_patches" {
+  count            = local.enabled ? 1 : 0
+  window_id        = aws_ssm_maintenance_window.scan_window[0].id
+  task_type        = "RUN_COMMAND"
+  task_arn         = "AWS-RunPatchBaseline"
+  priority         = var.task_scan_priority
+  service_role_arn = var.service_role_arn
+  max_concurrency  = var.max_concurrency
+  max_errors       = var.max_errors
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  targets {
+    key    = "WindowTargetIds"
+    values = [aws_ssm_maintenance_window_target.target_scan.*.id]
+  }
+
+  task_invocation_parameters {
+    run_command_parameters {
+      parameter {
+        name   = "Operation"
+        values = ["Scan"]
+      }
+      parameter {
+        name   = "RebootOption"
+        values = ["NoReboot"]
+      }
+      output_s3_bucket     = local.bucket_id
+      output_s3_key_prefix = "scaning"
+
+      service_role_arn = var.role_arn_for_notification
+
+      dynamic "notification_config" {
+        for_each = var.enable_notification_scan ? [1] : []
+        content {
+          notification_arn    = var.notification_arn
+          notification_events = var.notification_events
+          notification_type   = var.notification_type
+        }
+      }
+    }
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_ssm_maintenance_window_target" "target_scan" {
+  count         = local.enabled ? 1 : 0
+  window_id     = aws_ssm_maintenance_window.scan_window[0].id
+  resource_type = "INSTANCE"
+
+  dynamic "targets" {
+    for_each = toset(var.scan_maintenance_windows_targets)
+    content {
+      key    = targets.value.key
+      values = targets.value.values
+    }
+  }
+  dynamic "targets" {
+    for_each = length(var.scan_maintenance_windows_targets) == 0 ? [1] : []
+    content {
+      key    = "tag:Patch Group"
+      values = var.scan_patch_groups
+    }
+  }
+}
+
+# Maintenance Windows for patching
+
+resource "aws_ssm_maintenance_window" "install_window" {
+  count    = local.enabled ? 1 : 0
+  name     = "module.this.id"
```

Comment:
```suggestion
  name     = module.this.id
```

Reply:
The module creates many `aws_ssm_maintenance_window` resources.
Their names should be unique.

Use `null-label` with `attributes` to create different names for them

Add `install_window_label` label module.

Then use

```
name     = module. install_window_label.id
```